code/modules/food_and_drink/cakes.dm

```diff
@@ -24,62 +24,482 @@
+			if(topping && !(src.GetOverlayImage("[topping]"))) //actually adding the topping overlay to the cake
+				var/image/toppingoverlay = new /image('icons/obj/foodNdrink/food_dessert.dmi',"[topping]")
+				toppingoverlay.alpha = 255
+				src.UpdateOverlays(toppingoverlay,"[topping]")
+				user.u_equip(W)
+				qdel(W)
+				if(pendinglight)
+					src.ignite()
 				return
-			boutput(user, "<span class='notice'>You cut the cake into slices.</span>")
-			var/makeslices = src.amount
-			while (makeslices > 0)
-				var/obj/item/reagent_containers/food/snacks/cake/custom/P = new src.type(get_turf(src))
-				P.food_effects += src.food_effects
-				P.sliced = 1
-				P.amount = 1
-				P.w_class = 1
-				P.icon_state = "cake_slice"
-				P.quality = src.quality
-				P.name = src.name
-				P.desc = src.desc
-				P.food_color = src.food_color
-				P.update_icon(1)
-				P.icing_color = src.icing_color
-				P.update_icing(1)
-				src.reagents.trans_to(P, src.reagents.total_volume/makeslices)
-				P.pixel_x = rand(-6, 6)
-				P.pixel_y = rand(-6, 6)
-				makeslices--
-			qdel (src)
-		if(istype(W, /obj/item/reagent_containers/glass/bottle/icing) && !(icing) &&(!sliced) && (W.reagents.total_volume))
-			if(W.reagents.total_volume == 50)
-				boutput(user, "<span class='notice'>You add the icing to the cake.</span>")
-				icing = 1
-				src.desc += "<br>It has " + W.reagents.get_master_reagent_name() + " icing."
-				icing_color = W.reagents.get_master_color()
-				src.update_icing(0)
-				W.reagents.trans_to(src, W.reagents.total_volume)
+
+
+	attack_hand(mob/user as mob)
+		if((user.a_intent == INTENT_GRAB) && (src.clayer >1)) //removing layers from cakes.
+			var/obj/item/reagent_containers/food/snacks/cake/custom/s = new /obj/item/reagent_containers/food/snacks/cake/custom
+
+			src.reagents.trans_to(s,(src.reagents.total_volume/3))
+			for(var/i=1,i<=src.food_effects.len,i++)
+				if(src.food_effects[i] in s.food_effects)
+					continue
+				s.food_effects += src.food_effects[i]
+			s.quality = src.quality
+			s.food_color = src.food_color
+
+			build_cake(s,user,1)
+
+			if(istype(user,/mob/living/carbon/human))
+				user.put_in_hand_or_drop(s)
+			else
+				s.set_loc(get_turf(user))
+			if(src.litfam)
+				src.put_out()
+				s.ignite()
+		else if(user.a_intent == INTENT_DISARM) //blowing out candles
+			//check for candle
+			var/blowout
+			var/staticiterator = src.overlays.len
+			for(var/i=1,i<=staticiterator,i++)
+				if(src.sliced && ("[src.overlay_refs[i]]" == "slice-candle_lit"))
+					src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+					src.UpdateOverlays(new /image('icons/obj/foodNdrink/food_dessert.dmi',"slice-candle"), "slice-candle")
+					blowout = 1
+					break
+				if("[src.overlay_refs[i]]" == "cake[src.clayer]-candle_lit")
+					src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+					var/tag = "cake[src.clayer]-candle"
+					src.UpdateOverlays(new /image('icons/obj/foodNdrink/food_dessert.dmi',tag), tag)
+					blowout = 1
+					break
```

Comment:
```suggestion
      if(src.sliced && src.GetOverlayImage("slice-candle_lit"))
        src.ClearSpecificOverlays("slice-candle_lit")
        src.UpdateOverlays(image('icons/obj/foodNdrink/food_dessert.dmi',"slice-candle"), "slice-candle")
        blowout = 1
      else if(src.GetOverlayImage("cake[src.clayer]-candle_lit"))
        src.ClearSpecificOverlays("cake[src.clayer]-candle_lit")
        src.UpdateOverlays(image('icons/obj/foodNdrink/food_dessert.dmi',"cake[src.clayer]-candle_lit"), "cake[src.clayer]-candle")
        blowout = 1
```

Also in the future it'd probably make stuff simpler if you instead of having `-candle` and `-candle_lit` you had just one overlay slot and put the right image into it (removes the need of calling ClearSpecificOverlays etc.).

Reply:
Oh right...I made some additions in cleanup that made that obsolete...ill change that.

Reply:
*spins around in circles*

Reply:
i did a read through of all the times this is used and it honestly seems ok now that the other new optimizations in place, I don't think changing this would make the biggest difference performance-wise anymore. I'll close this for now unless someone thinks this is still a super relevant issue ❤︎